src/openscad.cc

```diff
@@ -1048,7 +1044,9 @@ int main(int argc, char **argv)
 				rc = info();
 			}
 			else {
-				rc = cmdline(deps_output_file, inputFiles[0], output_file, original_path, parameterFile, parameterSet, viewOptions, camera, export_format);
+				for(auto output_file : output_files) {
+					rc = cmdline(deps_output_file, inputFiles[0], output_file, original_path, parameterFile, parameterSet, viewOptions, camera, export_format);
```

Comment:
While this is safe (since it recompiles for every output file), it can also be quite inefficient.
Also keep in mind that this can yield unexpected results for designs that echo'es information based on the `$preview` variable.

I don't like the implications of this. My initial hunch would be to start only allowing .echo as the second output format (and perhaps other safe outputs).

Reply:
It might be a little bit inefficient, but AFAIK any GeometryCache will persist across those calls, as long as its the same process.

One of my motivations for this PR overall was that it could potentially be used to help test cache-related issues such as the one discussed here: https://github.com/openscad/openscad/issues/2842#issuecomment-469530635
By doing something like `-o foo.png -o foo.stl` forces preview before F6 for example, which would give different results (in that case) from only `-o foo.stl`.

I guess I'm not familiar with how echo from command line and `$preview` work together normally, so I don't know if this has potential to change it.